### PR TITLE
CheckboxListTile, RadioListTile, SwitchListTile

### LINKED ISSUE #254: Checkbox + label should be a widget
<a href="https://github.com/Hixie"></img></a> **Issue by [Hixie](https://github.com/Hixie)**
_Tuesday Nov 03, 2015 at 20:15 GMT_
_Originally opened as https://github.com/flutter/engine/issues/1925_

---

Any time you add a checkbox, you end up having to add a label also, and then hooking up the event handlers on both, and it's a pain. We should just have one widget that has a checkbox and a label.

Same for radio buttons, switches, etc.

### PR DESCRIPTION
**Pull request by Hixie**
cc @abarth 

Fixes https://github.com/flutter/flutter/issues/254

### CODE REVIEW

**Comment by Hixie** [comment]
(I plan on adding a MergeSemantics to these widgets.)

**Comment by abarth** [comment]
LGTM

**Comment by Hixie** [comment]
Ok, I made these accessible too.

**Comment by Hixie** [comment]
Will check in on green unless I hear otherwise.

File: packages/flutter/lib/src/material/radio_list_tile.dart [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+/// property, but the color used is that described by [activeColor], if any,
+/// defaulting to the accent color of the current [Theme]. No effort is made to
+/// coordinate the [selected] state and the [checked] state; to have the list
+/// tile appear selected when the radio button is the selected radio button, set
+/// [selected] to true when [value] matches [groupValue].
+///
+/// The radio button is shown on the left by default in left-to-right languages
+/// (i.e. the leading edge). This can be changed using [controlAffinity]. The
+/// [secondary] widget is placed on the opposite side. This maps to the
+/// [leading] and [trailing] properties of [ListTile].
```
Comment (Hixie) [inline packages/flutter/lib/src/material/radio_list_tile.dart]: need to be explicit and leading/trailing

**Pull request merged**